src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.ts

```diff
@@ -78,12 +80,41 @@ export class NavigationComponent implements OnInit, OnDestroy {
         this.showTopNotification('motdNotificationEnabled', _.isPlainObject(motd));
       })
     );
+    this.router.events.subscribe((event: Event) => {
+      if (event instanceof NavigationEnd) {
+        this.setDisplayedSubMenu();
+      }
+    });
   }
 
   ngOnDestroy(): void {
     this.subs.unsubscribe();
   }
 
+  setDisplayedSubMenu(): void {
+    if (
+      this.router.url.startsWith('/hosts') || this.router.url.startsWith('/inventory')
+      || this.router.url.startsWith('/monitor') || this.router.url.startsWith('/services')
+      || this.router.url.startsWith('/osd') || this.router.url.startsWith('/configuration')
+      || this.router.url.startsWith('/crush-map') || this.router.url.startsWith('/mgr-modules')
+      || this.router.url.startsWith('/logs') || this.router.url.startsWith('/monitoring')
```

Comment:
There is too much repetition here. Maybe:
```
url = '/monitor/blah'
let pages = ['/hosts','/inventory','/monitor','/services','/osd','/configuration','/crush-map','/mgr-modules','/logs','/monitoring']
let res = pages.some((page) => url.startsWith(page));
`

Reply:
I agree! Also if we change the url to start with `/cluster/hosts`, `/cluster/inventory` etc then we can just eliminate all the if conditions and just check 
```
if this.router.url.startsWith('/cluster/') 
    displaySubmenu = cluster`
```